corehq/apps/app_manager/models.py

```diff
@@ -5707,63 +5707,89 @@ def reapply_overrides(self):
+    if source_app.domain == domain:
+        app.family_id = source_app.origin_id
+
+    report_map = get_static_report_mapping(source_app.domain, domain)
+    _update_report_ids(app, report_map, source_app.domain, check_all_reports)
+
+    app.save_attachments(attachments)
+
+    _update_valid_domains_for_media(domain, app, request)
+
+    if not app.is_remote_app():
+        enable_usercase_if_necessary(app)
+
+    return app
+
+
+def _get_source_app(app_id_or_source):
     if isinstance(app_id_or_source, str):
         source_app = get_app(None, app_id_or_source)
     else:
         source_app = wrap_app(app_id_or_source)
-    source_domain = source_app.domain
-    source = source_app.export_json(dump_json=False)
+    return source_app
+
+
+def _get_attachments(source):
     try:
         attachments = source['_attachments']
     except KeyError:
         attachments = {}
-    finally:
-        source['_attachments'] = {}
-    if source_properties is not None:
-        for key, value in source_properties.items():
-            source[key] = value
-    cls = get_correct_app_class(source)
-    # Allow the wrapper to update to the current default build_spec
-    if 'build_spec' in source:
-        del source['build_spec']
-    app = cls.from_source(source, domain)
+
+    return attachments
+
+
+def _copy_app_from_source(domain, source):
+    app_class = get_correct_app_class(source)
+    app = app_class.from_source(source, domain)
     app.convert_build_to_app()
     app.date_created = datetime.datetime.utcnow()
     app.cloudcare_enabled = domain_has_privilege(domain, privileges.CLOUDCARE)
-    if source_domain == domain:
-        app.family_id = source_app.origin_id
 
-    report_map = get_static_report_mapping(source_domain, domain)
+    return app
+
+
+def _update_report_ids(app, report_map, domain, check_all_reports):
     if report_map:
         for module in app.get_report_modules():
             for config in module.report_configs:
                 try:
                     config.report_id = report_map[config.report_id]
                 except KeyError:
-                    if check_all_reports or config.report(source_domain).is_static:
+                    if check_all_reports or config.report(domain).is_static:
                         raise AppEditingError(
                             "Report {} not found in {}".format(config.report_id, domain)
                         )
 
-    app.save_attachments(attachments)
 
+def _update_valid_domains_for_media(app, domain_to_add, request):
     try:
         if not app.is_remote_app():
             for path, media in app.get_media_objects(remove_unused=True):
-                if domain not in media.valid_domains:
-                    media.valid_domains.append(domain)
+                if domain_to_add not in media.valid_domains:
+                    media.valid_domains.append(domain_to_add)
                     media.save()
     except ReportConfigurationNotFoundError:
```

Comment:
I don't see and can't imagine anywhere that a `ReportConfigurationNotFoundError` would be thrown while getting multimedia. I think this try/except can be removed, which would also let you remove the request param from `import_app`.

If you do this, it's worth spending a few minutes blaming this code to try to figure out how this code came to exist.

Reply:
It looks like it is there for good reason based on [this](https://github.com/dimagi/commcare-hq/pull/23991).

Reply:
Hmm, so the check was deliberately added, but I don't understand what's going on in that PR. Our support for mobile UCR + linked project spaces has changed recently - @mkangia do you have a sense of if this try/catch is still relevant? Even at the time the PR Graham linked was done, I don't see how `get_media_objects` would ever have thrown a `ReportConfigurationNotFoundError`

Reply:
Hey,

The stack trace in this ticket, https://dimagi-dev.atlassian.net/browse/ICDS-240, shares how that exception gets raised when fetching media objects. Looks like its just trying to load modules and hence its configs, ending up complaining about missing reports.

Considering a bit i have known about linked ucrs recently, i imagine, if someone copies an app that has mobile reports or mobile reports with UCRs and has not set linked ucrs, this error message should pop up.
And would be good to know that app would break. Plus, if we don't rescue it, the process if copy would fail.

Does that make sense?

Reply:
ohhhhh, thank you, I get it now, I hadn't read the stack trace closely enough. Yep, let's leave this try/catch alone.